### PR TITLE
Bugfix/zen 27876

### PR DESCRIPTION
**Pull request by lundybernard**

### CODE REVIEW

File: Products/ZenEvents/zentrap.py [hunk 1]
```diff
@@ -820,6 +748,107 @@ def cleanup(self):
         self.log.info(status)
 
 
+class Decoders:
+    '''methods to decode OID values
+    '''
```
Comment (jpeacock-zenoss) [inline Products/ZenEvents/zentrap.py]: Single quotes?  Double quotes.

File: Products/ZenEvents/zentrap.py [hunk 2]
```diff
@@ -820,6 +748,107 @@ def cleanup(self):
...
+            if value[0] in [0, 1, 2] and all(isinstance(i, int) for i in value):
+                return '.'.join(map(str, value))
+
+    @staticmethod
+    def ipaddress(value):
+        for version in [socket.AF_INET, socket.AF_INET6]:
+            try:
+                return socket.inet_ntop(version, value)
+            except ValueError:
+                None
```
Comment (jpeacock-zenoss) [inline Products/ZenEvents/zentrap.py]: While valid, this doesn't actually do what you think.  Function return `None` by default.  The idiom is to use `pass` as the exception handler.

File: Products/ZenEvents/zentrap.py [hunk 3]
```diff
@@ -820,6 +748,107 @@ def cleanup(self):
...
+    '''methods to decode OID values
+    '''
+
+    @staticmethod
+    def decode(value):
+        decoders = [Decoders.oid,
+                    Decoders.basestring,
+                    Decoders.ipaddress,
+                    Decoders.dateandtime,
+                    Decoders.encode_base64]
```
Comment (jpeacock-zenoss) [inline Products/ZenEvents/zentrap.py]: Need some commenting on the how the order of decoders is important.

File: Products/ZenEvents/zentrap.py (hunk 1 above)
Comment (lundybernard) [inline Products/ZenEvents/zentrap.py]: fixed

File: Products/ZenEvents/zentrap.py (hunk 3 above)
Comment (lundybernard) [inline Products/ZenEvents/zentrap.py]: added doc string

File: Products/ZenEvents/zentrap.py (hunk 2 above)
Comment (lundybernard) [inline Products/ZenEvents/zentrap.py]: fixed

**Comment by lundybernard** [comment]
retest this please

**Pull request merged**